### PR TITLE
Add Verify snapshot golden-master for dice notation (#173)

### LINKED ISSUE #173: [Maintenance] testing: Snapshot / approval testing for stable outputs (Verify framework)
## Scope

Where the library produces structured output (formatted strings, JSON, XML, CSV), snapshot testing via the [Verify](https://github.com/VerifyTests/Verify) framework catches accidental format drift that unit tests targeted assertions miss. The first run records the output as a `.verified.txt` file; subsequent runs diff against it.

## Acceptance criteria

- Verify package referenced in the test project.
- Snapshot tests cover every public method whose output is text-shaped, complex-shaped (records, anonymous objects), or has interesting edge-case formatting (culture-sensitive, timezone-sensitive, etc.).
- Snapshot files committed under `tests/.../Snapshots/`.
- CI verifies snapshots match; `.received.txt` files committed during local iteration are gitignored.

## Notes

- Tier-1 follow-up. Only meaningfully valuable where the library produces inspectable output; pure-data utility libraries can close as not-applicable.
- Originated as part of the IAsyncEnumerable-Extensions thorough-review pass; tracked fleet-wide.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### PR DESCRIPTION
**Pull request by Chris-Wolfgang**
Closes #173. Stacked on #228 (#183). Last of the vNext clean-doable stack.

## What

Adds snapshot/approval testing ([Verify.Xunit](https://github.com/VerifyTests/Verify)) with a single **golden-master** over a representative matrix of dice/die configurations:

```
Die(6)                   => d6
Dice(2,6,-1)             => 2d6-1
Dice([d6,d6,d4])         => 2d6+1d4
Dice([d10,d10,d6],-2)    => 2d10+1d6-2
...
```

A change to the `ToString` notation format surfaces **every** affected case in one snapshot diff — complementing (not replacing) the exact per-case `Assert.Equal` tests.

## Notes

- Guarded to **net10.0** (the format is TFM-independent and net10.0 runs in CI), so a single committed `.verified.txt` snapshot suffices.
- `*.received.*` is already gitignored; only the accepted `.verified.txt` is committed.
- Verify auto-disables its diff tool under CI, so the job won't hang.

188 tests pass on net10.0. No CHANGELOG entry (internal).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/Wolfgang.D20.Dice.Tests/Wolfgang.D20.Dice.Tests.Unit.csproj [hunk 1]
```diff
@@ -93,6 +93,8 @@
   	<PackageReference Include="xunit.extensibility.core" Version="2.9.3" />
   	<!-- Property-based tests (PropertyTests.cs) are guarded to net8.0+; CsCheck 4.x targets net8.0. -->
   	<PackageReference Include="CsCheck" Version="4.7.0" />
+  	<!-- Snapshot / approval tests (SnapshotTests.cs) are guarded to net8.0+. -->
+  	<PackageReference Include="Verify.Xunit" Version="31.12.5" />
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply tests/Wolfgang.D20.Dice.Tests/Wolfgang.D20.Dice.Tests.Unit.csproj]: Fixed in 0f31230 — scoped Verify.Xunit to a net10.0-only ItemGroup and corrected the comment (was net8.0+).

File: tests/Wolfgang.D20.Dice.Tests/SnapshotTests.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        Add("Dice(2,6)", new Dice(2, 6).ToString());
+        Add("Dice(2,6,+3)", new Dice(2, 6, 3).ToString());
+        Add("Dice(2,6,-1)", new Dice(2, 6, -1).ToString());
+        Add("Dice(1,20)", new Dice(1, 20).ToString());
+        Add("Dice([d6,d6,d4])", new Dice(new[] { new Die(6), new Die(6), new Die(4) }).ToString());
+        Add("Dice([d8,d4],+2)", new Dice(new[] { new Die(8), new Die(4) }, 2).ToString());
+        Add("Dice([d10,d10,d6],-2)", new Dice(new[] { new Die(10), new Die(10), new Die(6) }, -2).ToString());
+
+        return Verify(builder.ToString());
+    }
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply tests/Wolfgang.D20.Dice.Tests/SnapshotTests.cs:37]: Fixed in 0f31230 — snapshots now live under Snapshots/ via Verify UseDirectory; the verified file was moved there.

File: tests/Wolfgang.D20.Dice.Tests/SnapshotTests.Dice_notation_golden_master.verified.txt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿Die(6)                   => d6
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply tests/Wolfgang.D20.Dice.Tests/SnapshotTests.Dice_notation_golden_master.verified.txt]: Fixed in 0f31230 — re-saved the verified snapshot without the UTF-8 BOM (first line now starts directly with Die(6)).
